python/pyarrow/tests/parquet/test_basic.py

```diff
@@ -352,6 +352,37 @@ def test_byte_stream_split(use_legacy_dataset):
                          use_legacy_dataset=use_legacy_dataset)
 
 
+@parametrize_legacy_dataset
+def test_col_encoding(use_legacy_dataset):
+    arr_float = pa.array(list(map(float, range(100))))
+    arr_int = pa.array(list(map(int, range(100))))
+    mixed_table = pa.Table.from_arrays([arr_float, arr_int],
+                                       names=['a', 'b'])
+
+    # Check NONE col_encoding.
+    _check_roundtrip(mixed_table, expected=mixed_table, use_dictionary=False,
+                     col_encoding=None, use_legacy_dataset=use_legacy_dataset)
+
+    # Check "BYTE_STREAM_SPLIT" for column 'a' and "PLAIN" col_encoding for
+    # column 'b'.
+    _check_roundtrip(mixed_table, expected=mixed_table, use_dictionary=False,
+                     col_encoding={'a': "BYTE_STREAM_SPLIT", 'b': "PLAIN"},
+                     use_legacy_dataset=use_legacy_dataset)
+
+    # Check "RLE" for column 'a' and "BYTE_STREAM_SPLIT" col_encoding for
+    # column 'b'.
+    _check_roundtrip(mixed_table, expected=mixed_table,
+                     use_byte_stream_split=['a'],
+                     col_encoding={'a': "RLE", 'b': "BYTE_STREAM_SPLIT"},
```

Comment:
In this case we are overriding the `use_byte_stream_split`, I think? (since `col_encoding` is handled after `use_byte_stream_split`) In that case we should maybe raise an error that you can't set both at the same time?

Reply:
I was working in a way to override, yes (because of the description in [ARROW-13781](https://issues.apache.org/jira/browse/ARROW-13781) but am fine with raising an error.

Reply:
I think the description in ARROW-13781 ("supercede") might be about that with this new keyword, the existing `use_btye_stream_split` keyword is no longer needed since you can achieve the same with `col_encoding`. And indeed specifying it through `col_encoding` will ignore what was passed to `use_byte_stream_split`. So for users there is no need to specify both, in which case I would explicitly disallow that (or at least raise a warning), to avoid confusion.

Reply:
Ok, I get it.

I left `use_byte_stream_split` intact because you can set it to `True` and it is applied to all columns, which is handy. I will make a correction to connect `use_byte_stream_split` with `column_encoding` as we talked about.

Reply:
> I left `use_byte_stream_split` intact because you can set it to `True` and it is applied to all columns, which is handy.

This is maybe a useful feature to also add to `column_encoding`? Eg if you pass a single string, it is applied to all columns (using similar code as this is done for `use_byte_stream_split`), and otherwise a dict for per-column encoding.